codec/codectools/token.go

```diff
@@ -0,0 +1,78 @@
+package codectools
+
+import (
+	"fmt"
+
+	"github.com/ipld/go-ipld-prime"
+)
+
+type Token struct {
```

Comment:
I'm not a fan of the package name, because then the qualified names are a bit weird; `codectools.Token` sounds to me like it should be `codec.Token`.

I guess this is less important if the end user shouldn't see `codectools`, but then I would argue this should be an internal package.

Reply:
Agree, this is way too much of a mouthful, especially for the Token stuff.

Gonna run with it a while longer, because I've stacked up so more code atop this already, but some rename or reshuffle here would be an improvement.